.github/workflows/check-release.yml

```diff
@@ -0,0 +1,26 @@
+name: Check Release
+on:
+  push:
+    branches: [master, main]
+  pull_request:
+    branches:
+      - '*'
+
+permissions:
+  contents: write
+
+jobs:
+  check_release:
```

Comment:
This should be moved to `step_build` right?

Reply:
No, this workflow is for `jupyter-releaser`. Please look into its [docs](https://jupyter-releaser.readthedocs.io/en/latest/) to have a better understanding.

Reply:
Hmm, other than the risk of bumping the version, why shouldn't it be in the `step_build`. The purpose for that one is to simulate a packaging step, which this one seems to do as well. In the `publish.yml` then the artifacts of `pipx build` and/or this job should be used.

From what I read in the implementation `check-release` is just a combination of all the other github actions, which in principle is more robust for using when releasing. What are your plans on how to implement this?

Reply:
Exactly `check-release` simulates all the steps in actual `jupyter-releaser` workflow using mock GH instance. It does more than _just_ building packages like simulating changelog generation, packaging, making release with mock instance, _etc_. As this workflow is very _specific_ to `jupyter-releaser`, in my opinion it makes sense to keep it in a separate file. I mean, we have split the existing _dense_ workflow file into logical step files for better readability. I think adding it to `setp_build` will again mishmash workflow files. 

My original plan was to do include this `check-release` workflow _simulation_ in the repo and give an idea to maintainers on what exactly it is doing. I guess @mwouts is taking of releases and if he wishes to use `jupyter-releaser` workflows for publishing packages, we can add the actual workflows in a separate PR.

Reply:
My view is that `step_*` workflows are not a single-job files, but more like logically separated ones. E.g. `step_pre-commit` should also have the `npm` pre-commits. In that sense it makes sense to have them in `step_build` (or renamed to something else) because in essence that step is about testing the packaging, and if this one includes changelog generation and all, all the better.

Main concern is the `on:` and linking the workflows one to another. But for this first-iteration we can keep it here since it still under consideration on how to implement.